server/routes/polls/polls.route.js

```diff
@@ -2,82 +2,98 @@ const express = require('express');
 const router = express.Router();
 const Poll = require('../../models/polls');
 const upload = require('./utils');
+const passport = require('passport');
 
 // @route POST /polls
 // @desc To add a new poll
 // @access private
 // @params title, userId, imagess x 2
-router.post('/', upload.any(), (req, res) => {
-  const errors = [];
-  if (!req.body.title) errors.push({ title: 'Invalid title' });
-  if (!req.body.userId) errors.push({ userId: 'Invalid user id' });
-  if (req.files.length != 2)
-    errors.push({ images: 'minimum two images required' });
+router.post(
+  '/',
+  passport.authenticate('jwt', { session: false }),
+  upload.any(),
+  (req, res) => {
+    const errors = [];
+    if (!req.body.title) errors.push({ title: 'Invalid title' });
+    if (!req.body.userId) errors.push({ userId: 'Invalid user id' });
+    if (req.files.length != 2)
+      errors.push({ images: 'minimum two images required' });
 
-  if (errors.length > 0) return res.status(400).json({ error: errors });
-  const poll = new Poll({
-    title: req.body.title,
-    userId: req.body.userId,
-  });
-  // Trying to save the poll
-  try {
-    poll.save();
-  } catch (error) {
-    return res.status(400).json({ error: error.message });
-  }
-  // Saving file location to the poll object
-  try {
-    req.files.map((item) => {
-      poll.addImage(item.location);
+    if (errors.length > 0) return res.status(400).json({ error: errors });
+    const poll = new Poll({
+      title: req.body.title,
+      userId: req.body.userId,
```

Comment:
We are still not checking if req.body.userId is the same as req.user.id

Take a look at: https://github.com/hatchways/team-eagle/blob/dev/server/config/passport.js

That middleware will fetch the user and save it as req.user (see http://www.passportjs.org/packages/passport-jwt/ for more detail)